api/client/service/update.go

```diff
@@ -122,8 +117,8 @@ func updateService(spec *swarm.ServiceSpec, flags *pflag.FlagSet) error {
 	updateString(flagName, &spec.Name)
 	updateLabels(flags, &spec.Labels)
 	updateString("image", &cspec.Image)
-	updateSlice("command", &cspec.Command)
-	updateSlice("arg", &cspec.Command)
+	updateStringToSlice(flags, "command", &cspec.Command)
```

Comment:
> Is this the behavior you had in mind for command / args?

This is the correct behavior and allows backwards compat with entrypoint/cmd.

`Command` and `Args` are simply vectors of string, as is supported in any shell. A `Command` may have arguments that come along with it. Imagine `["myapp", "serve"]` as being a possible entrypoint in an image. When we execute a containers, we effectively append `Args` to `Command`. So, the actual command becomes ` `["myapp", "serve", "args"]`, if `Args` is `["args"]`.

> I'm just following what swarmkit supports. I think it would make more sense for swarmkit to return to the entrypoint/command behaviour we have with docker run.

`Command`/`Args` _is_ the correct approach and is an industry standard. _Every other orchestration_ system does it like this.